src/variant.cpp

```diff
@@ -737,7 +737,12 @@ void from_variant( const variant& var,  std::vector<char>& vo )
 {
    const auto& str = var.get_string();
    FC_ASSERT( str.size() <= 2*MAX_SIZE_OF_BYTE_ARRAYS ); // Doubled because hex strings needs two characters per byte
-   vo.resize( str.size() / 2 ); {
+   if( str.find("0x") != string::npos )
+      str = str.substr(2);
+   if( str.size() % 2 )
+      str.insert(0, "0");
+   vo.resize( str.size() / 2 );
```

Comment:
`from_hex` already supports odd-number length hex strings if given sufficient space in the `vector<char>`. You can achieve this without inserting a '0' character to the end of `str` by instead simply changing this line to: `vo.resize( (str.size()+1) / 2 );`.

However, I think it may be a better idea for this function to just assert that the string length is not odd to avoid potential ambiguity.